lib/benchee/benchmark/measure/memory.ex

```diff
@@ -44,17 +52,19 @@ defmodule Benchee.Benchmark.Measure.Memory do
   defp return_memory({memory_usage, result}) when memory_usage < 0, do: {nil, result}
   defp return_memory({memory_usage, result}), do: {memory_usage, result}
 
-  defp measure_memory(fun, tracer) do
-    word_size = :erlang.system_info(:wordsize)
-    {:garbage_collection_info, heap_before} = Process.info(self(), :garbage_collection_info)
+  defp measure_memory(fun, tracer, parent) do
+    :erlang.garbage_collect()
+    send(tracer, :begin_collection)
+    receive do
+      :ready -> nil
+    end
     result = fun.()
-    {:garbage_collection_info, heap_after} = Process.info(self(), :garbage_collection_info)
-    mem_collected = get_collected_memory(tracer)
-
-    memory_used =
-      (total_memory(heap_after) - total_memory(heap_before) + mem_collected) * word_size
-
-    {memory_used, result}
+    send(parent, {:result, result})
```

Comment:
I'm doing this so the result can be garbage collected on the following line after the final reference to it is released and included in the final total of memory usage.